### PR TITLE
Update setup.py and buildscripts for dependency requirements

### PR DESCRIPTION
**Pull request by sklam**
Constrain:

- llvmlite version to the minor release 
- numpy version >= 1.15 (np1.11 for building is still supported)
- scipy version >= 1.0.0

### CODE REVIEW

**Comment by sklam** [comment]
Testing on `numba_smoketest_cpu8`

**Comment by sklam** [comment]
Started `numba_smoketest_cpu_9`

**Comment by sklam** [comment]
Trying it in a full dag run: `numba_90`

File: azure-pipelines.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ trigger:
 
 variables:
   # Change the following along with adding new TEST_START_INDEX.
-  TEST_COUNT: 18
+  TEST_COUNT: 16
```
Comment (stuartarchibald) [inline azure-pipelines.yml]: At some point we should just write a script that generates these Azure yml files as having to keep twiddling the start indexes/count is not ideal.

**Comment by stuartarchibald** [comment]
> Thanks for the patch. On inspection this appears to do as described. I think the docs etc will need an update, is this for another issue or to go on here? Thanks.

From an out of band conversation, this will be done in a follow up PR.

**Pull request merged**